nucypher/characters/control/serializers.py

```diff
@@ -0,0 +1,219 @@
+import json
+from abc import ABC
+from base64 import b64decode, b64encode
+from json import JSONDecodeError
+from typing import Callable
+
+import maya
+
+import nucypher
+from nucypher.characters.control.specifications import CharacterSpecification
+
+
+class CharacterControlSerializer(ABC):
+
+    class SerializerError(ValueError):
+        """Invalid data for I/O serialization or deserialization"""
+
+
+class CharacterControlJSONSerializer(CharacterControlSerializer):
+
+    def __call__(self, data, specification: tuple, *args, **kwargs):
```

Comment:
Am I right that you just call `serializer()` to both serialize and deserialize, and decide what to do depending on the `data` type? Just not sure if it can create errors which are hard to catch

Reply:
Yes, that is the way this is designed, but I am just a s happy with them as separate methods.

Reply:
I'd like to hear inputs from others on this one. But yeah, I think, `load` / `dump` or whatever would be more explicit and readable, although more characters

Reply:
Perhaps another post-PR optimization, if that's OK.

Reply:
https://github.com/nucypher/nucypher/issues/814